src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -160,7 +168,41 @@ static IContainSagaData DeepClone(IContainSagaData source)
 
             public IContainSagaData GetSagaCopy()
             {
-                return DeepClone(data);
+                var canBeClonedShallowly = canBeClonedShallowlyCache.GetOrAdd(data.GetType(), CanBeMemberwiselyCloned);
+                return canBeClonedShallowly ? memberwiseClone(data) : DeepClone(data);
+            }
+
+            static bool CanBeMemberwiselyCloned(Type type)
```

Comment:
How about `CanBeShallowCloned` or since this method is called from `GetSagaCopy`, how about `CanBeShallowCopied` ?